From: Andreas Henriksson <andreas@fatal.se>
Date: Sun, 11 Aug 2019 13:03:13 +0200
Subject: build: create daemon storage dirs on install

Upon initial install iwd fails to start because the service file
says ReadWritePaths=/var/lib/iwd while that directory doesn't
(yet) exist. (This commit also fixes the same problem for ead.)

Addresses: https://bugs.debian.org/934194
Reported-by: Stephen Gelman <ssgelm@debian.org>
(cherry picked from commit 5a96c11664eb553bc28a2142af382b190254edbb)
---
 Makefile.am | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index cb28f0c..c620da2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -580,6 +580,14 @@ SED_PROCESS = $(AM_V_GEN)$(MKDIR_P) $(dir $@) && $(SED) \
 %.1: %.txt
 	$(AM_V_GEN)$(A2X) --doctype manpage --format manpage $(srcdir)/$<
 
+install-data-local:
+if DAEMON
+	$(MKDIR_P) -m 700 $(DESTDIR)$(daemon_storagedir)
+endif
+if WIRED
+	$(MKDIR_P) -m 700 $(DESTDIR)$(wired_storagedir)
+endif
+
 clean-local:
 	-rm -f unit/cert-*.pem unit/cert-*.csr unit/cert-*.srl
 
